tools/privacyidea-cron

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# 2018-06-29 Friedrich Weber <friedrich.weber@netknights.it>
+#            Implement periodic task runner
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import sys
+
+__doc__ = """
+This script is meant to be invoked periodically by the system cron daemon.
+It runs periodic tasks that are specified in the database.
+"""
+__version__ = "0.1"
+
+from datetime import datetime
+
+import dateutil
+from flask_script import Manager
+
+from privacyidea.app import create_app
+from privacyidea.lib.periodictask import get_scheduled_periodic_tasks, set_periodic_task_last_run, get_periodic_task_by_name, \
+    execute_task
+from privacyidea.lib.error import ParameterError
+
+app = create_app(config_name='production', silent=True)
+manager = Manager(app)
+
+
+def get_node_name(node):
+    """
+    Determine the node name. If no node name is given, read it from the app config.
+    Raise a ParameterError if no node name could be determined.
+    :param node: node name given by the user (can be None)
+    :return:
+    """
+    if node is None:
+        if "PI_AUDIT_SERVERNAME" in app.config:
+            return app.config["PI_AUDIT_SERVERNAME"]
+        else:
+            raise ParameterError("No PI_AUDIT_SERVERNAME specified in pi.cfg, use --node to pass a node name")
+    else:
+        return node
+
+
+def run_task_on_node(ptask, node):
+    """
+    Run a periodic task (given as a dictionary) on the given node.
+    In case of success, write the last successful run to the database. Catch any exceptions.
+    :param ptask: task as a dictionary
+    :param node: Node name
+    """
+    try:
+        print u"Running {!r} ...".format(ptask["name"]),
```

Comment:
for python 3 compat we should already write ``print(value)``  not ``print value``.

There are several occurances.